### ISSUE TITLE
Add Support for Temporal Tables

### ISSUE THREAD

**Issue by NatevdhNHA** [tSQLt-org/tSQLt#40]
I wanted to test if a Table had an Audit setup on it and some of the tables are now using temporal tables to do that auditing.

I did see that Michael Hotek had built his own version of it over in this google group https://groups.google.com/forum/#!topic/tsqlt/nsHpez08huA

He implemented it by creating a separate stored procedure `tSQLt.FakeTableTemporal` but would it make more sense to extend `tsqlt.FakeTable` to support it instead?

Not sure which is the best way to go but wanted to get the issue in

**Comment by Bryant4044** [comment]
Extending FakeTable to handle temporal might not be that easily done. With temporal you want the ability to seed history data which means you need to defer making the faked table system-versioned. Adding those features via FakeTable directly would add complexity to FakeTable not needed for run-of-the-mill tables.

In my organization we have a companion schema for internal tSQLt extensions. I've added Michael's code plus a wrapper around one of his private methods to make it simple for developers to handle faking the history table.

**Comment by dkultasev** [comment]
We have our own implementation of faking temporal tables. I can't say that
it is very hard to implement. The biggest problem was making different
values for period columns it has the same value for all records per
transaction. But even that can be manually updated later on.

**Comment by Bryant4044** [comment]
Agreed, it isn't all that difficult. Retrofitting FakeTable might be more effort than necessary and _potentially_ breaks a lot more code. Simply having a FakeTableTemporal method would streamline development and testing.

**Comment by RFluegel** [comment]
I worked on it before I was aware of the mentioned solution it is quite simple and does not interfere much with the implementation of tSQLt. It works for my usecases and might be useful for someone else. Therefore I put it here.

[ExtendtSQLt_Documentation.docx](https://github.com/tSQLt-org/tSQLt/files/5577220/ExtendtSQLt_Documentation.docx)
[tSQLExtension.zip](https://github.com/tSQLt-org/tSQLt/files/5577221/tSQLExtension.zip)

**Comment by fdw** [comment]
@RFluegel We'd like to use your extension, so first: Thank you! 🙂

Unfortunately, you haven't really attached a license, just a limited warranty. So, what are the conditions for its use?

**Comment by RFluegel** [comment]
It is nice that the extension is useful.
  
  
   I did not think about a license.
  
  
   You can use the extension in any way you want.
  
  
   
    Fabian Winter ***@***.***> hat am 23.06.2022 10:30 geschrieben:
   
   
    
   
   
    
   
   
   @RFluegel We'd like to use your extension, so first: Thank you! 🙂
   Unfortunately, you haven't really attached a license, just a limited warranty. So, what are the conditions for its use?
